From 216e08f5593b3d32628f58c1d4d6a6f49e94eb57 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 07 Apr 2019 15:12:55 +0300
Subject: [PATCH] [SW-0] pll: update hl_pll_cfg.base type

It is simple u64 and not a pointer this avoids redundand casting
Also make it cleaner for use with io operations (readl/writel)

Change-Id: Ib7752fad3711dac85b22d381a066c248e86584a5
---

diff --git a/goya/pll.c b/goya/pll.c
index df855cf..54a03e5 100644
--- a/goya/pll.c
+++ b/goya/pll.c
@@ -48,7 +48,7 @@
 
 struct hl_pll_cfg {
 	char	*name;
-	void	*base;
+	u64	base;
 	u32	nr;
 	u32	nf;
 	u32	od;
@@ -58,15 +58,15 @@
 
 static struct hl_pll_cfg goya_pll_cfg[] = {
 	{
-		.name = "CPU", .base = (void *)mmCPU_PLL_BASE,
+		.name = "CPU", .base = mmCPU_PLL_BASE,
 		.nr = 2, .nf = 159, .od = 1, .nb = 79,
 		.div_factor = { 0, -1, 1, -1 }
 	}, {
-		.name = "IC", .base = (void *)mmIC_PLL_BASE,
+		.name = "IC", .base = mmIC_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 15, .nb = 31,
 		.div_factor = { 0, 1, 1, 2 }
 	}, {
-		.name = "MME", .base = (void *)mmPSOC_MME_PLL_BASE,
+		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 15, .nb = 31,
 		.div_factor = { 0, 1, 2, 8 }
 	},
@@ -168,7 +168,7 @@
 static void hl_pll_bypass(struct hl_pll_cfg *cfg, bool bypass)
 {
 	int sel;
-	void *reg;
+	u64 reg;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
 		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
@@ -188,7 +188,7 @@
 static void hl_pll_div_init(struct hl_pll_cfg *cfg)
 {
 	int sel;
-	void *div_base;
+	u64 div_base;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
 		if (cfg->div_factor[sel] == -1)
@@ -241,7 +241,7 @@
 	for (i = 0; i < pll_num; i++) {
 		/* DIV0 is the undivided PLL output */
 		bval = hl_pll_bypass_get(hl_pll_cfg + i, HL_PLL_DIV0);
-		if ((u64)hl_pll_cfg[i].base == mmCPU_PLL_BASE &&
+		if (hl_pll_cfg[i].base == mmCPU_PLL_BASE &&
 				bval != HL_PLL_REF_CLK &&
 				bval != HL_PLL_DIV_REF)
 			continue;
